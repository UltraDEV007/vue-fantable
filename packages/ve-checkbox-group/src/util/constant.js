// prefix
export const PREFIX_CLS = 've-checkbox-group-'

// emit events
export const EMIT_EVENTS = {
  // on checked change
  ON_CHECKED_CHANGE: 'on-checked-change',
}

// comps name
export const COMPS_NAME = {
  VE_CHECKBOX: 'VeCheckbox',
  VE_CHECKBOX_GROUP: 'VeCheckboxGroup',
}
