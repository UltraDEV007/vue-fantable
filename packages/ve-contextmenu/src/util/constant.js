// prefix
export const PREFIX_CLS = 've-contextmenu-'

// comps name
export const COMPS_NAME = {
  VE_CONTEXTMENU: 'VeContextmenu',
}

// emit events
export const EMIT_EVENTS = {
  ON_NODE_CLICK: 'on-node-click',
}

// init data
export const INIT_DATA = {
  // PARENT_ID: 0,
  PARENT_DEEP: -1,
}

// contextmenu types
export const CONTEXTMENU_NODE_TYPES = {
  // Separator
  SEPARATOR: 'SEPARATOR',
}

// instance methods
export const INSTANCE_METHODS = {
  HIDE_CONTEXTMENU: 'hideContextmenu',
}
