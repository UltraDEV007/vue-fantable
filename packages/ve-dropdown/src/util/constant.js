// prefix
export const PREFIX_CLS = 've-dropdown-'

// emit events
export const EMIT_EVENTS = {
  FILTER_RESET: 'on-filter-reset',
  FILTER_CONFIRM: 'on-filter-confirm',
  VISIBLE_CHANGE: 'on-dropdown-visible-change',
  ITEM_SELECT_CHANGE: 'on-item-select-change',
}

// comps name
export const COMPS_NAME = {
  VE_DROPDOWN: 'VeDropdown',
}
