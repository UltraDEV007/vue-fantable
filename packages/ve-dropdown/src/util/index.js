/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return 've-dropdown-' + cls
}
