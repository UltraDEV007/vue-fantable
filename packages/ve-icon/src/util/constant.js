// prefix
export const PREFIX_CLS = 've-icon-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_ICON: 'VeIcon',
}
