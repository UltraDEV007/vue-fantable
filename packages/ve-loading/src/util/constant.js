// prefix
export const PREFIX_CLS = 've-loading-'

// emit events
export const EMIT_EVENTS = {}

// comps name
export const COMPS_NAME = {
  VE_LOADING: 'VeLoading',
  VE_LOADING_PLANE: 'VeLoadingPlane',
  /* VE_LOADING_CHASE: "VeLoadingChase", */
  VE_LOADING_BOUNCE: 'VeLoadingBounce',
  VE_LOADING_WAVE: 'VeLoadingWave',
  VE_LOADING_PULSE: 'VeLoadingPulse',
  VE_LOADING_FLOW: 'VeLoadingFlow',
  /* VE_LOADING_SWING: "VeLoadingSwing", */
  /*  VE_LOADING_CIRCLE: "VeLoadingCircle",
VE_LOADING_CIRCLE_FADE: "VeLoadingCircleFade", */
  VE_LOADING_GRID: 'VeLoadingGrid',
  VE_LOADING_FOLD: 'VeLoadingFold',
}

// spin name
export const SPIN_NAMES = {
  PLANE: 'plane',
  GRID: 'grid',
  WAVE: 'wave',
  FLOW: 'flow',
  BOUNCE: 'bounce',
  PULSE: 'pulse',
}
