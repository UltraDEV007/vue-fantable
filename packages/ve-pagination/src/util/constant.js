// prefix
export const PREFIX_CLS = 've-pagination-'

// locale comp name
export const LOCALE_COMP_NAME = 'pagination'

// emit events
export const EMIT_EVENTS = {
  PAGE_NUMBER_CHANGE: 'on-page-number-change',
  PAGE_SIZE_CHANGE: 'on-page-size-change',
}

// comps name
export const COMPS_NAME = {
  VE_PAGINATION: 'VePagination',
}
