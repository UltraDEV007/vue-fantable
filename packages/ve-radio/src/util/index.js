import { PREFIX_CLS } from './constant'

/*
 * @clsName
 * @desc  get class name
 * @param {string} cls - class
 */
export function clsName(cls) {
  return PREFIX_CLS + cls
}
