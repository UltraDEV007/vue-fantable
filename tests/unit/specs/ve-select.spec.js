import { mount } from "@vue/test-utils";
import veSelect from "@/ve-select";
import { later } from "../util";

describe("veSelect", () => {
    // select items
    const SELECT_ITEMS = [
        { value: 0, label: "Apple" },
        { value: 1, label: "Orange", selected: true },
        { value: 2, label: "Banana" },
    ];

    it("render by value prop", () => {
        const wrapper = mount({
            render() {
                return <veSelect value={SELECT_ITEMS} placeholder="姓名" />;
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render by isMultiple prop", () => {
        const wrapper = mount({
            render() {
                return (
                    <veSelect
                        value={SELECT_ITEMS}
                        placeholder="姓名"
                        isMultiple
                    />
                );
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("render by isInput prop", () => {
        const wrapper = mount({
            render() {
                return (
                    <veSelect value={SELECT_ITEMS} placeholder="姓名" isInput />
                );
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it("width prop", () => {
        const wrapper = mount(veSelect, {
            propsData: {
                value: SELECT_ITEMS,
                width: 120,
            },
        });

        expect(
            wrapper.find(".ve-dropdown-dt-selected").attributes("style"),
        ).toBe("width: 120px;");
    });

    it("isMultiple prop", () => {
        const wrapper = mount(veSelect, {
            propsData: {
                value: SELECT_ITEMS,
                isMultiple: true,
            },
        });

        expect(wrapper.findAll(".ve-dropdown-items-multiple").length).toBe(3);
    });

    it("isInput prop", () => {
        const wrapper = mount(veSelect, {
            propsData: {
                value: SELECT_ITEMS,
                isInput: true,
            },
        });

        expect(wrapper.find(".ve-select-input").exists()).toBe(true);
    });

    it("on-select-change emit event", async () => {
        const wrapper = mount(veSelect, {
            propsData: {
                value: SELECT_ITEMS,
            },
        });

        wrapper.findAll(".ve-dropdown-items-li").at(1).trigger("click");

        expect(wrapper.emitted("on-select-change").length).toEqual(1);
        expect(
            wrapper.emitted("on-select-change")[0][0].find((x) => x.selected)
                .label,
        ).toBe("Orange");
    });
});
