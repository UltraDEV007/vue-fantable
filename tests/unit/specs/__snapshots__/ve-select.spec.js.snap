// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`veSelect render by isInput prop 1`] = `
<dl class="ve-dropdown ve-select">
  <dt class="ve-dropdown-dt"><a class="ve-dropdown-dt-selected" style="width: 90px;"><span><input placeholder="姓名" type="text" class="ve-select-input"><i class="ve-icon iconfont-vet icon-vet-bottom-arrow ve-select-toggle-icon"></i></span></a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Apple</a></li>
            <li class="ve-dropdown-items-li active"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Orange</a></li>
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Banana</a></li>
          </div>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;

exports[`veSelect render by isMultiple prop 1`] = `
<dl class="ve-dropdown ve-select">
  <dt class="ve-dropdown-dt"><a class="ve-dropdown-dt-selected" style="width: 90px;"><span><span class="ve-select-selected-span">Orange</span><i class="ve-icon iconfont-vet icon-vet-bottom-arrow ve-select-toggle-icon"></i></span></a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value="Apple"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Apple</span></label></li>
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content ve-checkbox-checked"><input type="checkbox" class="ve-checkbox-input" value="Orange"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Orange</span></label></li>
            <li class="ve-dropdown-items-multiple ve-dropdown-items-li ve-dropdown-items-li-a-left"><label class="ve-checkbox" style="display: inline-block;"><span class="ve-checkbox-content"><input type="checkbox" class="ve-checkbox-input" value="Banana"><span class="ve-checkbox-inner"></span></span><span class="ve-checkbox-label">Banana</span></label></li>
          </div>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;

exports[`veSelect render by value prop 1`] = `
<dl class="ve-dropdown ve-select">
  <dt class="ve-dropdown-dt"><a class="ve-dropdown-dt-selected" style="width: 90px;"><span><span class="ve-select-selected-span">Orange</span><i class="ve-icon iconfont-vet icon-vet-bottom-arrow ve-select-toggle-icon"></i></span></a></dt>
  <div style="display: none;">
    <dd class="ve-dropdown-dd">
      <ul class="ve-dropdown-items" style="min-width: 90px; max-width: Infinitypx;">
        <div>
          <div class="ve-dropdown-items-warpper" style="max-height: 1000px;">
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Apple</a></li>
            <li class="ve-dropdown-items-li active"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Orange</a></li>
            <li class="ve-dropdown-items-li"><a href="javascript:void(0);" class="ve-dropdown-items-li-a ve-dropdown-items-li-a-left">Banana</a></li>
          </div>
        </div>
      </ul>
    </dd>
  </div>
</dl>
`;
